### ISSUE TITLE
connection timeout on first try

### ISSUE THREAD

**Issue by bonesoul** [agenda/agenda#770]
on windows, agenda is having difficulties connecting to mongodb. after a re-try it's all good again;

```
   [ 'MongoNetworkError: failed to connect to server [localhost:27017] on first connect [MongoNetworkError: connection 0 to localhost:27017 timed out]',
     '    at Pool.<anonymous> (E:\\Coding\\hypetrain\\node_modules\\agenda\\node_modules\\mongodb-core\\lib\\topologies\\server.js:564:11)',
     '    at Pool.emit (events.js:197:13)',
     '    at Pool.EventEmitter.emit (domain.js:439:20)',
     '    at Connection.<anonymous> (E:\\Coding\\hypetrain\\node_modules\\agenda\\node_modules\\mongodb-core\\lib\\connection\\pool.js:317:12)',
     '    at Object.onceWrapper (events.js:285:13)',
     '    at Connection.emit (events.js:197:13)',
     '    at Connection.EventEmitter.emit (domain.js:439:20)',
     '    at Socket.<anonymous> (E:\\Coding\\hypetrain\\node_modules\\agenda\\node_modules\\mongodb-core\\lib\\connection\\connection.js:257:10)',
     '    at Object.onceWrapper (events.js:285:13)',
     '    at Socket.emit (events.js:197:13)',
     '    at Socket.EventEmitter.emit (domain.js:439:20)',
     '    at Socket._onTimeout (net.js:447:8)',
     '    at listOnTimeout (timers.js:327:15)',
     '    at processTimers (timers.js:271:5)' ] }
```

**Comment by bonesoul** [comment]
still having the issue.

**Comment by lgcantarelli** [comment]
+1

**Comment by Adithya188288** [comment]
Same here in agenda version ^4.1.0. But you can load agenda again in MongoDB reconnection event and it works.

**Comment by hunganhAtZehitomo** [comment]
Has anyone been able to look into this and find the root cause?

**Comment by hunganhAtZehitomo** [comment]
bump

**Comment by pdfowler** [comment]
@hunganhAtZehitomo i don't believe this is an agenda specific issue, as I've seen it frequently over the years in my local dev environments. Most often, I see it when connecting to a DB instance that has been idle for a long time. I've looked into it a few times and I think there are some mongo connection settings you can try adjusting, but a google search shows that the solutions are as varied as everyone's env.

**Comment by hunganhAtZehitomo** [comment]
FYI I finally resolved the issue (at least for my environment). Since I use docker for dev envs and I have relatively small amount of RAM (8gb), I had to add options: { useUnifiedTopology: true } to the Agenda instantiation.
TLDR; If your machine sees high RAM usage, add options: { useUnifiedTopology: true } to new Agenda.
e.g
`const agenda = new Agenda({ db: { address, collection: "your_collection", options: { useUnifiedTopology: true } }, processEvery: "40 seconds"})`
